diff -up qttools-everywhere-src-6.0.0-beta4/cmake/FindWrapLibClang.cmake.1~ qttools-everywhere-src-6.0.0-beta4/cmake/FindWrapLibClang.cmake
--- qttools-everywhere-src-6.0.0-beta4/cmake/FindWrapLibClang.cmake.1~	2020-11-15 23:01:07.789414428 +0100
+++ qttools-everywhere-src-6.0.0-beta4/cmake/FindWrapLibClang.cmake	2020-11-15 23:01:52.387151627 +0100
@@ -58,7 +58,7 @@ if(TARGET libclang AND ((TARGET clang-cp
         qt_internal_disable_find_package_global_promotion(Threads::Threads)
         # lupdate must also link to LLVM when using clang-cpp
         set(__qt_clang_genex_condition "$<AND:$<TARGET_EXISTS:clang-cpp>,$<TARGET_EXISTS:LLVM>>")
-        set(__qt_clang_genex "$<IF:${__qt_clang_genex_condition},clang-cpp;LLVM,clangHandleCXX>")
+        set(__qt_clang_genex "$<IF:${__qt_clang_genex_condition},clang-cpp;LLVM,clangHandleCXX> -lclangAST -lclangBasic -lclangFrontend -lclangLex -lclangSerialization -lclangTooling -lLLVMSupport")
         target_link_libraries(WrapLibClang::WrapLibClang
             INTERFACE libclang
             ${__qt_clang_genex}
