diff -up qttools-everywhere-src-6.7.0-beta1/src/qdoc/qdoc/src/qdoc/clang/AST/QualTypeNames.h.omv~ qttools-everywhere-src-6.7.0-beta1/src/qdoc/qdoc/src/qdoc/clang/AST/QualTypeNames.h
--- qttools-everywhere-src-6.7.0-beta1/src/qdoc/qdoc/src/qdoc/clang/AST/QualTypeNames.h.omv~	2024-01-27 18:23:08.693808649 +0100
+++ qttools-everywhere-src-6.7.0-beta1/src/qdoc/qdoc/src/qdoc/clang/AST/QualTypeNames.h	2024-01-27 18:23:37.642098007 +0100
@@ -438,7 +438,7 @@ inline QualType getFullyQualifiedType(Qu
   // elaborated type.
   Qualifiers PrefixQualifiers = QT.getLocalQualifiers();
   QT = QualType(QT.getTypePtr(), 0);
-  ElaboratedTypeKeyword Keyword = ETK_None;
+  ElaboratedTypeKeyword Keyword = ElaboratedTypeKeyword::None;
   if (const auto *ETypeInput = dyn_cast<ElaboratedType>(QT.getTypePtr())) {
     QT = ETypeInput->getNamedType();
     assert(!QT.hasLocalQualifiers());
@@ -469,7 +469,7 @@ inline QualType getFullyQualifiedType(Qu
         Ctx, QT.getTypePtr(), WithGlobalNsPrefix);
     QT = QualType(TypePtr, 0);
   }
-  if (Prefix || Keyword != ETK_None) {
+  if (Prefix || Keyword != ElaboratedTypeKeyword::None) {
     QT = Ctx.getElaboratedType(Keyword, Prefix, QT);
   }
   QT = Ctx.getQualifiedType(QT, PrefixQualifiers);
