diff -up qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.cpp.omv~ qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.cpp
--- qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.cpp.omv~	2022-09-02 18:00:24.203265182 +0200
+++ qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.cpp	2022-09-02 18:05:48.430538571 +0200
@@ -156,14 +156,23 @@ void LupdatePPCallbacks::SourceRangeSkip
 // To list the included files
 void LupdatePPCallbacks::InclusionDirective(clang::SourceLocation /*hashLoc*/,
     const clang::Token & /*includeTok*/, clang::StringRef /*fileName*/, bool /*isAngled*/,
-    clang::CharSourceRange /*filenameRange*/, const clang::FileEntry *file,
+    clang::CharSourceRange /*filenameRange*/,
+#if LLVM_VERSION_MAJOR < 15
+    const clang::FileEntry *file,
+#else
+    llvm::Optional<clang::FileEntryRef> file,
+#endif
     clang::StringRef /*searchPath*/, clang::StringRef /*relativePath*/,
     const clang::Module */*imported*/, clang::SrcMgr::CharacteristicKind /*fileType*/)
 {
     if (!file)
         return;
 
+#if LLVM_VERSION_MAJOR < 15
     clang::StringRef fileNameRealPath = file->tryGetRealPathName();
+#else
+    clang::StringRef fileNameRealPath = file->getFileEntry().tryGetRealPathName();
+#endif
     if (!LupdatePrivate::isFileSignificant(fileNameRealPath.str()))
         return;
 
diff -up qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.h.omv~ qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.h
--- qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.h.omv~	2022-09-02 17:55:05.113063670 +0200
+++ qttools-everywhere-src-6.4.0-beta4/src/linguist/lupdate/lupdatepreprocessoraction.h	2022-09-02 17:56:03.546538990 +0200
@@ -51,7 +51,12 @@ private:
     void SourceRangeSkipped(clang::SourceRange sourceRange, clang::SourceLocation endifLoc) override;
     void InclusionDirective(clang::SourceLocation /*hashLoc*/, const clang::Token &/*includeTok*/,
                             clang::StringRef /*fileName*/, bool /*isAngled*/,
-                            clang::CharSourceRange /*filenameRange*/, const clang::FileEntry *file,
+                            clang::CharSourceRange /*filenameRange*/,
+#if LLVM_VERSION_MAJOR < 15
+			    const clang::FileEntry *file,
+#else
+			    llvm::Optional<clang::FileEntryRef> file,
+#endif
                             clang::StringRef /*searchPath*/, clang::StringRef /*relativePath*/,
                             const clang::Module */*imported*/,
                             clang::SrcMgr::CharacteristicKind /*fileType*/) override;
