diff -up qt6-qttools-6.11.0/src/qdoc/cmake/QDocConfiguration.cmake.4~ qt6-qttools-6.11.0/src/qdoc/cmake/QDocConfiguration.cmake
--- qt6-qttools-6.11.0/src/qdoc/cmake/QDocConfiguration.cmake.4~	2026-03-10 06:07:41.000000000 +0100
+++ qt6-qttools-6.11.0/src/qdoc/cmake/QDocConfiguration.cmake	2026-03-31 17:18:18.834418485 +0200
@@ -8,7 +8,7 @@ set(QDOC_MINIMUM_CLANG_VERSION "17")
 
 # List of explicitly supported Clang versions for QDoc
 set(QDOC_SUPPORTED_CLANG_VERSIONS
-    "21.1" "20.1" "19.1" "18.1" "17.0.6"
+    "23.1" "22.1" "21.1" "20.1" "19.1" "18.1" "17.0.6"
 )
 
 # Check for QDoc coverage dependencies
diff -up qt6-qttools-6.11.0/src/qdoc/qdoc/src/qdoc/clangcodeparser.cpp.4~ qt6-qttools-6.11.0/src/qdoc/qdoc/src/qdoc/clangcodeparser.cpp
--- qt6-qttools-6.11.0/src/qdoc/qdoc/src/qdoc/clangcodeparser.cpp.4~	2026-03-10 06:07:41.000000000 +0100
+++ qt6-qttools-6.11.0/src/qdoc/qdoc/src/qdoc/clangcodeparser.cpp	2026-03-31 17:17:43.259453302 +0200
@@ -29,6 +29,8 @@
 #include <QtCore/qtextstream.h>
 #include <QtCore/qvarlengtharray.h>
 
+#undef emit // Clang 22.1+ defines a BoundTypeDiagnoser::emit method
+
 #include <clang-c/Index.h>
 
 #include <clang/AST/Decl.h>
@@ -42,7 +44,11 @@
 #include <clang/Lex/Lexer.h>
 #include <llvm/Support/Casting.h>
 
+#if LIBCLANG_VERSION_MAJOR >= 22
+#include <clang/AST/QualTypeNames.h>
+#else
 #include "clang/AST/QualTypeNames.h"
+#endif
 #include "template_declaration.h"
 
 #include <cstdio>
