diff -uraN qttools-everywhere-src-6.3.0/cmake/FindWrapLibClang.cmake omv-qttools-everywhere-src-6.3.0/cmake/FindWrapLibClang.cmake
--- qttools-everywhere-src-6.3.0/cmake/FindWrapLibClang.cmake	2022-04-05 23:41:58.000000000 +0200
+++ omv-qttools-everywhere-src-6.3.0/cmake/FindWrapLibClang.cmake	2022-05-02 16:41:39.965233611 +0200
@@ -59,6 +59,7 @@
         # lupdate must also link to LLVM when using clang-cpp
         set(__qt_clang_genex_condition "$<AND:$<TARGET_EXISTS:clang-cpp>,$<TARGET_EXISTS:LLVM>>")
         set(__qt_clang_genex "$<IF:${__qt_clang_genex_condition},clang-cpp;LLVM,clangHandleCXX>")
+        set(__qt_clang_genex "$<IF:${__qt_clang_genex_condition},clang-cpp;LLVM,clangHandleCXX> -lclangAST -lclangBasic -lclangFrontend -lclangLex -lclangSerialization -lclangTooling -lLLVMSupport")
         target_link_libraries(WrapLibClang::WrapLibClang
             INTERFACE libclang
             "${__qt_clang_genex}"
